components/widget/widget_button.lua

```diff
@@ -0,0 +1,77 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Button
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+
+local Widget = Lua.import('Module:Widget')
+local Link = Lua.import('Module:Widget/Link')
+local Div = Lua.import('Module:Widget/Div')
+
+---@class ButtonWidgetParameters: WidgetParameters
+---@field title string?
+---@field link string?
+---@field linktype 'internal'|'external'|nil
+---@field variant 'primary'|'secondary'|'ghost'|nil
+---@field size 'sm'|'md'|'lg'|nil
+
+---@class ButtonWidget: Widget
+---@operator call(ButtonWidgetParameters): ButtonWidget
+---@field title string?
+---@field link string?
+---@field linktype 'internal'|'external'
+---@field variant 'primary'|'secondary'|'ghost'
+---@field size 'sm'|'md'|'lg'
+
+local Button = Class.new(
+	Widget,
+	function(self, input)
+		self.title = input.title
+		self.link = input.link
+		self.linktype = input.linktype or 'internal'
+		self.variant = input.variant or 'primary'
+		self.size = input.size or 'md'
+	end
+)
+
+---@param children string[]
+---@return string
+function Button:make(children)
+	--- MW Parser does not allowed the <button> tag, so we use a div for now instead
+	local button = mw.html.create('div'):addClass('btn'):addClass('btn-new')
+	button:attr('title', self.title)
+	button:attr('aria-label', self.title)
```

Comment:
aria-label on a button is only needed when the button doesn't have text (for example only an icon) so might want to set that as optional

Reply:
self.title is optional, if `nil/null` is pased into it the attr is a no-op